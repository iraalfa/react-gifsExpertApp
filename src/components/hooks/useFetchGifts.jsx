import React from 'react';
import { getGifts } from '../../helpers/getGifts';

/* custom hook para cargar automaticamente las busquedas de gifts
   lo que hacemos es que cuando se cargue el componente por primera vez
   hace una peticion para obtener los gifts y los coloca en las images */

export const useFetchGifts = (category) => {
    const [state, setState ] = React.useState({
        data: [],
        loading: true,
    })
/*
    en el useEffect = llamada fetch en el componente getGifts.
                      con .then le pasamos las imgs que recibimos de category y las
                      pasamos al state en data y poniendo loading en false, porque ya se ha hecho la carga.
*/
    React.useEffect(() => {
        getGifts(category).then(imgs => {
            setState({
                 data:imgs,
                 loading: false
            })
        })
    })

    // React.useEffect(() => {
    //     getGifts(category)
    //     .then(imgs => {
    //             setTimeout(() => {
    //                 setState({
    //                     data: imgs,
    //                     loading: false,
    //                 })
    //             }, 3000)
    //     });
    // }, [category ])

    return state
}