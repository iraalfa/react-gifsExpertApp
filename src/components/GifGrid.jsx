import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifts } from './hooks/useFetchGifts'
import GiftsGridItem from './GiftsGridItem'

const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifts(category)

    console.log(loading);

    return (
        <>
        <h3 className="animate__animated.animate__fadeIn">{category}</h3>

            {/* {loading ? 'Cargando...' : 'data cargada'} */}
            { loading && <p className="animate__animated.animate__flash">Cargando</p>}

         <div className="card-grid">
                {
                    images.map(img => {
                        return <GiftsGridItem
                                    key={img.id}
                                    {...img}
                                />
                    })
                }
        </div>
          {/* images.map(img => {
                     return   <GiftsGridItem
                                key={img.id}
                                img={img}
                               />
                    }) */}
                {/* images.map( ({id, title}) => ( <li key={id}>{title}</li>)) */}
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifGrid
