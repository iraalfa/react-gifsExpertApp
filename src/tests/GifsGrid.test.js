import { shallow } from "enzyme";
import GifGrid from "../components/GifGrid"
import { useFetchGifts } from "../components/hooks/useFetchGifts";
jest.mock("../components/hooks/useFetchGifts")


describe('pruebas en <GifGrid />', () => {

    useFetchGifts.mockReturnValue({
        data: [],
        loading: true,
    })

    const category = 'luigi'
    let wrapper = shallow(<GifGrid category={category}/>)

    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifts', () => {
       const gifts = [{
           id: 'abc',
           url: 'http://example.com',
           title: 'titulo de prueba'
       }]
          expect(wrapper).toMatchSnapshot()
          expect(wrapper.find('p').exists()).toBe(false)
          expect(wrapper.find('GifGridItem').length).toBe(gifts.length)
    });

});
