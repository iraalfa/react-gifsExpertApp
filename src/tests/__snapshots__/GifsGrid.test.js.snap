// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated.animate__fadeIn"
  >
    luigi
  </h3>
  <p
    className="animate__animated.animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated.animate__fadeIn"
  >
    luigi
  </h3>
  <p
    className="animate__animated.animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
